enum.h

```diff
@@ -217,6 +217,7 @@ struct optional {
 
 namespace _enum {
 
+#pragma GCC diagnostic ignored "-Weffc++"
```

Comment:
This is caused by a mistake on my part. `operator =` should be "normal", i.e. have return type `const _eat_assign&` and return `*this`.